test/unit/test_vector.c

```diff
@@ -2,6 +2,12 @@
 #include <r_vector.h>
 #include "minunit.h"
 
+#if __WORDSIZE == 32
```

Comment:
I dont see the need for this defines here. Why not just use always 4?

Reply:
If the initial capacity differs in 32 and 64bit versions of it something wrong is going on somewhere else i think

Reply:
> If the initial capacity differs in 32 and 64bit versions of it something wrong is going on somewhere else i think

I updated the description above (forgot to reference line 5, which says the choice of initial capacity is for glibc optimization.
imho if it brings trouble elsewhere, the problem comes from the user of r_vector, rather than r_vector itself. We shouldn't write code whose correctness depends on specific implementation of a library. Ironically this unit test fails on arm32 because it depends on specific implementation :-)

Reply:
if its a glibc optimization this test may probably fail in other libcs like for Android or so. so its a bad idea to workaround this in the testsuite. i prefer not to include workarounds

Reply:
It won't fail, since the choice of initial capacity is for every libc but is only optimal on glibc. But I'll remove the test.